twitchio/websocket.py

```diff
@@ -171,8 +187,9 @@ async def _keep_alive(self):
                         continue
                     task = asyncio.create_task(self._process_data(event))
                     task.add_done_callback(partial(self._task_callback, event))  # Process our raw data
+                    self._background_tasks.append(task)
 
-        asyncio.create_task(self._connect())
+        await self._connect()
```

Comment:
why?

Reply:
Don't we want to make sure here that a (re)connect should definitely take place at this point?
Furthermore, we spare ourselves an attribute to hold a reference to the task.

Reply:
Don't we want to make sure here that a (re)connect should definitely take place at this point?
Furthermore, we spare ourselves an attribute to hold a reference to the task.

Reply:
@IAmTomahawkx what do you think?

Reply:
creating the task will cause a reconnect to take place when the event loop is ready. The keep alive function shouldn't be responsible for the lifespan of the connect call, as the connect function will cancel the keep alive, causing it to cancel itself at the next context switch

Reply:
Thanks for explaning, done